torch_geometric/utils/homophily.py

```diff
@@ -63,11 +85,26 @@ def homophily(edge_index: Adj, y: Tensor, batch: OptTensor = None,
         else:
             return scatter_mean(out, batch[col], dim=0)
 
-    else:
+    elif method == 'node':
         out = torch.zeros(row.size(0), device=row.device)
         out[y[row] == y[col]] = 1.
         out = scatter_mean(out, col, 0, dim_size=y.size(0))
         if batch is None:
             return float(out.mean())
         else:
             return scatter_mean(out, batch, dim=0)
+
+    else:
+        c = y.squeeze().max() + 1
+        counts = y.unique(return_counts=True)[1]
+        proportions = counts.float() / y.shape[0]
+        h = homophily(edge_index, y, batch=y, method='edge')
+
+        out = 0
+        for k in range(c):
+            class_add = torch.clamp(h[k] - proportions[k], min=0)
+            if not torch.isnan(class_add):
+                out += class_add
```

Comment:
Can we replace this with something like `torch.clamp(h-proportions).sum()`, I guess you do need to check for `nan` when nodes are isolated, could we could use `torch.nan_to_num` for that.

Reply:
The refactoring can be done without any issues. For isolated nodes, we can also call the `remove_isolated_nodes` helper before computing the homophily measure. What do you think?

Reply:
Ah now I don't think we need either. `h = homophily(edge_index, y, batch=y, method='edge')` this returns `return scatter_mean(out, batch[col], dim=0)` and as `batch = y` isolated nodes are either going to have h as 0 or h is not going to be as long as we expect( when the isolated nodes class is `max(y)`). For the first case we don't need the check for nan and for the second we just need to append some zeros to `h`
What do you think.

Reply:
Exactly. One can omit appending zeros to `h` by making use of `bincount` rather than `unique(return_counts=True)`.